zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/storage/BackupStorageUtil.java

```diff
@@ -143,7 +143,8 @@ public static void createFile(File file, boolean overwriteIfExist) throws IOExce
       return new File[0];
     }
     FilenameFilter fileFilter = (dir, name) -> name.startsWith(prefix);
-    return directory.listFiles(fileFilter);
+    File[] files = directory.listFiles(fileFilter);
+    return files == null ? new File[0] : files;
```

Comment:
I would recommend using a constant `NO_FILES = {};` to represent it, as it is also used in multiple places.

And why do we need to specially treat null? `null` is returned when the path is not directory or IO exception occurs. 
1. I think you might want to check if the path is directory `isDirectory()`
2. For IO exception case, you might want to handle it, eg. by retrying or else.

Reply:
Thanks for your suggestions. I will change it to a constant.
Regarding to the null treatment, please refer to [this PR](https://github.com/linkedin/zookeeper/pull/25) where we had some discussions around this and decided to specially treat it to make the logic consistent with other existing open source backup storage implementations.